packages/create-react-app/createReactApp.js

```diff
@@ -606,7 +606,7 @@ function isSafeToCreateProjectIn(root, name) {
 }
 
 function checkIfOnline(useYarn) {
-  if (!useYarn) {
+  if (!useYarn || process.env.https_proxy) {
```

Comment:
I'm not sure if skipping this check when proxy is defined is the best course of action -- can we try to resolve the URL through the proxy instead?